portfolio/src/main/webapp/script.js

```diff
@@ -1,21 +1,52 @@
 /**
  * Delete from datastore
  */
-async function deleteData() {
+function deleteData() {
   fetch('/delete-data', {method:'POST'}).then(() => {
       document.getElementById("alert-card").style.display = 'none';
       getData();
   });
 }
 
-async function deleteBuffer() {
-  fetch('/checklogin').then((response) => response.text())
-  .then((htmlContent) => {
+function deleteBuffer() {
+  fetch('/checklogin').then((response) => response.json())
+  .then((responseJson) => {
     const alertCard = document.getElementById('alert-card');
-    alertCard.innerHTML = htmlContent;
+    alertCard.innerHTML = '';
+    let message = document.createElement('p');
+
+    if (responseJson.loggedIn) {
+      const email = responseJson.email; 
+      if (email === 'sherryshi2001@gmail.com' || email === 'shershi@google.com' ||
+          email === 'alfredh@google.com' || email === 'ricazhang@google.com') {
```

Comment:
Sorry for the confusion on this. I think the server should still do the validation of email address because otherwise someone could go into the Javascript and just change the emails to be validated.

Reply:
oh wow yeah... okay, will fix.

Reply:
Will do this in the next PR